modules/jupyter-widget/src/utils.js

```diff
@@ -18,23 +18,26 @@ export function hideMapboxCSSWarning() {
 }
 
 export function updateDeck(inputJSON, {jsonConverter, deckConfig}) {
-  const results = jsonConverter.convertJsonToDeckProps(inputJSON);
+  const results = jsonConverter.convert(inputJSON);
   deckConfig.setProps(results);
 }
 
 export function initDeck({mapboxApiKey, container, jsonInput}, onComplete, handleClick) {
   require(['mapbox-gl', 'h3', 'S2'], mapboxgl => {
-    require(['deck.gl'], deckgl => {
+    require(['deck.gl', 'loaders.gl/core', 'loaders.gl/csv'], (deckgl, loadersCore, loadersCsv) => {
       try {
-        const layersDict = {};
-        const layers = Object.keys(deckgl).filter(
-          x => x.indexOf('Layer') > 0 && x.indexOf('_') !== 0
+        // Filter down to the deck.gl classes of interest
+        const classesDict = {};
+        const classes = Object.keys(deckgl).filter(
+          x => (x.indexOf('Layer') > 0 || x.indexOf('View') > 0) && x.indexOf('_') !== 0
         );
-        layers.map(k => (layersDict[k] = deckgl[k]));
+        classes.map(k => (classesDict[k] = deckgl[k]));
+
+        loadersCore.registerLoaders([loadersCsv.CSVLoader]);
```

Comment:
Does jupyter ever pass in CSV? My understanding is that it's all serialized JSON right now. If we are expanding the formats supported then we should go all the way to using binary.

Reply:
You can create the deck/JSON configuration in Python and only execute standalone deck.gl on the frontend without ever loading data to the Python backend.

Reply:
This is for letting deck.gl layers load CSV urls. E.g. the heatmap-3d/hexagon-layer example